osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,207 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <AccCtrl.h>
+#include <Aclapi.h>
+#include <map>
+#include <string>
+#include <vector>
+#include <windows.h>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem.hpp>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace alg = boost::algorithm;
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+// helper function to get access mode string
+std::string accessModeToStr(const _ACCESS_MODE& accessMode) {
+  switch (accessMode) {
```

Comment:
Why case/switch in a function here? Why not make this a statically defined `std::map<unsigned long, std::string>`? You'll save on resources allocated and it should be faster to access a map than call a function.

Reply:
`switch-case` is a bit faster.
But using `map` is much more readable. In this way I'd suggest to use `unordered_map`, it is notable faster.